File: easy/38.外观数列.py
#
# @lc app=leetcode.cn id=38 lang=python3
#
# [38] 外观数列
#

# @lc code=start
class Solution:
    def countAndSay(self, n: int) -> str:
        result = '1'
        for _ in range(n - 1):
            p, s = 0, ''
            for i, char in enumerate(result):
                if result[p] != char:
                    s += str(i - p) + result[p]
                    p = i
            result = s + str(len(result) - p) + result[p]
        return result


# @lc code=end

File: easy/38.外观数列.js
/*
 * @lc app=leetcode.cn id=38 lang=javascript
 *
 * [38] 外观数列
 */

// @lc code=start
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
  if (n === 1) {
    return '1';
  }
  let result = countAndSay(n - 1);
  let pointer = 0;
  let s = '';
  for (let i = 0; i < result.length; i += 1) {
    if (result[pointer] !== result[i]) {
      s += (i - pointer) + result[pointer];
      pointer = i;
    }
  }
  return s + (result.length - pointer) + result[pointer];
};
// @lc code=end

